tests/slsa_analyzer/test_git_url.py

```diff
@@ -122,36 +118,110 @@ def test_get_remote_vcs_url() -> None:
+    [
+        (
+            """
+            [git_service.github]
+            domain = github.com
+
+            [git_service.gitlab.public]
+            domain = gitlab.com
+            """,
+            {"github.com", "gitlab.com"},
+        ),
+        (
+            """
+            [git_service.gitlab.public]
+            domain = gitlab.com
+
+            [git_service.gitlab.private]
+            domain = internal.gitlab.org
+            """,
+            {"gitlab.com", "internal.gitlab.org"},
+        ),
+    ],
+)
+def test_get_allowed_git_service_domains(
+    config_input: str,
+    expected_allowed_domain_set: set[str],
+) -> None:
+    """Test the get allowed git service domains function."""
+    config = configparser.ConfigParser()
+    config.read_string(config_input)
+    assert set(git_url.get_allowed_git_service_domains(config)) == expected_allowed_domain_set
+
+
+@pytest.mark.parametrize(
+    ("user_config_input", "expected_allowed_domain_set"),
+    [
+        pytest.param(
+            # The current behavior is: we always enable GitHub and public GitLab by default.
+            # User config cannot disable either of the two.
+            """
+            [git_service.github]
+            domain = github.com
+            """,
+            {"github.com", "gitlab.com"},
+            id="Only GitHub in user config",
+        ),
+        pytest.param(
+            """
+            [git_service.gitlab.private]
+            domain = internal.gitlab.org
+            """,
+            {"github.com", "gitlab.com", "internal.gitlab.org"},
+            id="Private GitLab in user config",
+        ),
+    ],
+)
+def test_get_allowed_git_service_domains_with_override(
+    user_config_input: str,
+    expected_allowed_domain_set: set[str],
+    tmp_path: Path,
+) -> None:
+    """Test the get allowed git service domains function, in multi-config files scenario."""
+    user_config_path = os.path.join(tmp_path, "config.ini")
+    with open(user_config_path, "w", encoding="utf-8") as user_config_file:
+        user_config_file.write(user_config_input)
+    # We don't have to worry about modifying the ``defaults`` object causing test
+    # pollution here, since we reload the ``defaults`` object before every test with the
+    # ``setup_test`` fixture.
+    load_defaults(user_config_path)
+
+    assert set(git_url.get_allowed_git_service_domains(defaults)) == expected_allowed_domain_set
+
+
+def test_get_remote_vcs_url_with_user_defined_allowed_domains(tmp_path: Path) -> None:
+    """Test the vcs URL validator method with user-defined allowed domains."""
+    url = "https://internal.gitlab.org/org/name"
+    assert git_url.get_remote_vcs_url(url) == ""
+
+    user_config_path = os.path.join(tmp_path, "config.ini")
+    with open(user_config_path, "w", encoding="utf-8") as user_config_file:
+        user_config_file.write(
+            """
+                               [git_service.gitlab.private]
```

Comment:
Not sure, why `black` hasn't fixed the indentation here?

Reply:
Hmm, this is actually the way `black` fixes it. I think initially I typed

```
        user_config_file.write("""
                               [git_service.gitlab.private]
                               ...
                               """)
```
I guess I could have helped `black` a bit here. Fixed in 965282fca5bf228b12b154755cf15f3ec3acfa2a.